src/parse/parse-auth-request-query-params.ts

```diff
@@ -0,0 +1,258 @@
+import { VectorOfTrust, vtrValidator } from "../validators/vtr-validator";
+import { VALID_OIDC_PROMPTS } from "../constants";
+import { ParseRequestError } from "../errors/parse-request-error";
+import { logger } from "../logger";
+import { Request } from "express";
+import { AuthoriseRequestError } from "../errors/authorise-request-error";
+import { BadRequestError } from "../errors/bad-request-error";
+import { Config } from "../config";
+import { areScopesValid } from "../validators/scope-validator";
+import { areClaimsValid } from "../validators/claims-validator";
+import { MissingParameterError } from "../errors/missing-parameter-error";
+import { DuplicateParameterError } from "../errors/duplicate-parameter-error";
+
+export type ParsedAuthRequestQueryParams = {
+  response_type: string;
+  redirect_uri: string;
+  client_id: string;
+  state: string;
+  nonce: string;
+  scope: string[];
+  claims: string[];
+  vtr: VectorOfTrust[];
+  prompt: string[];
+  ui_locales: string[];
+};
+
+export const parseAuthQueryParams = (
+  queryParams: Request["query"]
+): ParsedAuthRequestQueryParams => {
+  if (isEmptyRequest(queryParams)) {
+    throw new MissingParameterError(
+      "Invalid Request: No Query parameters present in request"
+    );
+  }
+  const config = Config.getInstance();
+
+  if (!areDuplicatesPresent(queryParams)) {
+    throw new DuplicateParameterError(
+      "Invalid Request: Parameter(s) present more than once"
+    );
+  }
+
+  if (!queryParams.client_id) {
+    throw new MissingParameterError(
+      "Invalid Request: Missing client_id parameter"
+    );
+  }
+
+  if (!queryParams.redirect_uri) {
+    throw new MissingParameterError(
+      "Invalid Request: Invalid redirect_uri parameter"
+    );
+  }
+
+  if (!isValidUri(queryParams.redirect_uri)) {
+    throw new MissingParameterError(
+      "Invalid Request: Invalid redirect_uri parameter"
+    );
+  }
+
+  if (!queryParams.response_type) {
+    throw new ParseRequestError(
+      "Invalid Request: Invalid response_type parameter",
+      queryParams.redirect_uri,
+      queryParams.client_id
+    );
+  }
+
+  if (!queryParams.scope) {
+    throw new ParseRequestError(
+      "Invalid Request: Missing scope parameter",
+      queryParams.redirect_uri,
+      queryParams.client_id
+    );
+  }
+
```

Comment:
response type is another example of something we don't explicitly ask for, but should check we give the same response for